Project/Source/Components/UI/ComponentProgressBar.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "Components/Component.h"
+
+#include "Math/float4.h"
+#include "Math/float2.h"
+
+class ComponentTransform2D;
+
+// Component that renders an Image on a Quad
+class ComponentProgressBar : public Component {
+public:
+	REGISTER_COMPONENT(ComponentProgressBar, ComponentType::PROGRESS_BAR, false);
+
+	~ComponentProgressBar(); // Destructor
+
+	void Init() override;							// Inits the component
+	void Update() override;							// Update
+	void OnEditorUpdate() override;					// Input for parameters
+	void Save(JsonValue jComponent) const override; // Serializes object
+	void Load(JsonValue jComponent) override;		// Deserializes object
+	void DuplicateComponent(GameObject& owner) override;
+
+	void SetValue(float v);
+	void SetFillPos(float fillPos);
+	void SetMin(float m);
+	void SetMax(float n);
+	void SetFillDir(bool fillDir);
+	void SetFillDirBottom(bool fillDir);
+	void SetFillDirTop(bool fillDir);
+
+public:
+	float value = .5f;
+
+private:
+	GameObject* background = nullptr;
+	GameObject* fill = nullptr;
+
+	ComponentTransform2D* rectBack = nullptr;
+	ComponentTransform2D* rectFill = nullptr;
+
+	float3 backPos = float3::zero;
+	float2 backSize = float2::zero;
```

Comment:
background and fill should also be serialized and loaded as GameObject UIDs.
I'm not sure if the rects and backPos/Size are also needed.

Cannot test in Game without them!